run-directive/action.yml

```diff
@@ -0,0 +1,58 @@
+name: Run Directive
+run-name: test
+
+inputs:
+  repository:
+    description: 'Github repo'
+    required: true
+  hash:
+    description: 'Github commit hash to checkout'
+    required: false
+    default: ''
+  working-directory:
+    description: 'Name of component directory'
+    required: true
+  config:
+    description: 'Config JSON file'
+    required: true
+  stage:
+    description: 'Name of stage'
+    required: true
+  directive:
+    description: 'An npm directive. e.g. deploy, test'
+    required: true
+
+jobs:
+  run-directive:
+    name: Run Directive
+    using: "composite"
+
+    steps:
+      - name: Setup Environment
+        uses: Brightspace/bdp-ci-action/setup-env@master
+
+      - name: Checkout branch
+        uses: Brightspace/third-party-actions@actions/checkout
+        with:
+          repository: ${{ inputs.repository }}
+          ref: ${{ inputs.hash }}
+          fetch-depth: 0
+
+      - name: Use Node.js
+        uses: Brightspace/third-party-actions@actions/setup-node
+        with:
+          node-version-file: .nvmrc
+
+      - name: Add CodeArtifact npm registry
+        uses: Brightspace/codeartifact-actions/npm/add-registry@main
+        with:
+          auth-token: ${{secrets.CODEARTIFACT_AUTH_TOKEN}}
```

Comment:
I'm not sure if we need to setup node or code artifact. I figure since we might be deploying or running npm ci, these might need to be run before some directives and there is no harm to always running them.

Reply:
I think we do 👍🏻 Good call